sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -62,6 +62,8 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] {
   /** Returns a Seq of the children of this node */
   def children: Seq[BaseType]
 
+  lazy val childrenSet:Set[TreeNode[_]] = children.toSet
```

Comment:
Good idea.

Thanks

Mick

> On 12 Jun 2015, at 05:34, Wenchen Fan <notifications@github.com> wrote:
> 
> In sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala <https://github.com/apache/spark/pull/6673#discussion_r32289725>:
> 
> > @@ -62,6 +62,8 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] {
> >    /** Returns a Seq of the children of this node */
> >    def children: Seq[BaseType]
> >  
> > +  lazy val childrenSet:Set[TreeNode[_]] = children.toSet
> how about naming it containsChild so that we can use it like containsChild(arg)?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/apache/spark/pull/6673/files#r32289725>.
>